toolchain/check/context.h

```diff
@@ -21,6 +21,9 @@ namespace Carbon::Check {
 // Context and shared functionality for semantics handlers.
 class Context {
  public:
+  using DiagnosticEmitter = Carbon::DiagnosticEmitter<Parse::Node>;
```

Comment:
Perhaps a better solution would be to put the diagnostic mechanism in a namespace, so this would be `DiagnosticEmitter` versus `Diagnostic::Emitter<...>`?

Reply:
Sure, that's fair, and out-of-scope here.